src/writer.rs

```diff
@@ -0,0 +1,373 @@
+    writer.write_f32::<B>(header.scl_inter).unwrap();
+    writer.write_i16::<B>(header.slice_end).unwrap();
+    writer.write_u8(header.slice_code).unwrap();
+    writer.write_u8(header.xyzt_units).unwrap();
+    writer.write_f32::<B>(header.cal_max).unwrap();
+    writer.write_f32::<B>(header.cal_min).unwrap();
+    writer.write_f32::<B>(header.slice_duration).unwrap();
+    writer.write_f32::<B>(header.toffset).unwrap();
+    writer.write_i32::<B>(header.glmax).unwrap();
+    writer.write_i32::<B>(header.glmin).unwrap();
+
+    writer.write_all(&header.descrip).unwrap();
+    writer.write_all(&header.aux_file).unwrap();
+    writer.write_i16::<B>(header.qform_code).unwrap();
+    writer.write_i16::<B>(header.sform_code).unwrap();
+    for f in &[header.quatern_b, header.quatern_c, header.quatern_d,
+               header.quatern_x, header.quatern_y, header.quatern_z] {
+        writer.write_f32::<B>(*f).unwrap();
+    }
+    for f in header.srow_x.iter().chain(&header.srow_y).chain(&header.srow_z) {
+        writer.write_f32::<B>(*f).unwrap();
+    }
+    writer.write_all(&header.intent_name).unwrap();
+    writer.write_all(&header.magic).unwrap();
+
+    // Empty 4 bytes after the header
+    writer.write_u32::<B>(0).unwrap();
+
+    // We finally write the data.
+    // Like NiBabel, we iterate by "slice" to improve speed and use less memory
+
+    // Need the transpose for fortran used in nifti file format.
+    let data = data.t();
+
+    let arr_len = data.subview(Axis(0), 0).len();
+    let resolution: usize = header.dim[1..(header.dim[0] + 1) as usize]
+        .iter().map(|d| *d as usize).product();
+    let nb_bytes = resolution * header.bitpix as usize / 8;
+    let dim0_size = data.shape()[0];
+
+    // `1.0x + 0.0` would give the same results, but we avoid a lot of divisions
+    let slope = if header.scl_slope == 0.0 { 1.0 } else { header.scl_slope };
+    if slope != 1.0 || header.scl_inter != 0.0 {
+        let slope = A::from_f32(slope).unwrap();
+        let inter = A::from_f32(header.scl_inter).unwrap();
+        for arr_data in data.axis_iter(Axis(0)) {
+            let arr_data = arr_data.sub(inter).div(slope);
+            let arr_data = arr_data.into_shape(arr_len).unwrap();
+            let arr_data = arr_data.as_slice().unwrap();
+
+            let buffer = to_bytes(&arr_data, nb_bytes / dim0_size);
+            writer.write_all(&buffer).unwrap();
+        }
+    } else {
+        for arr_data in data.axis_iter(Axis(0)) {
+            // We need to own the data because of the into_shape() for `C` ordering.
+            let arr_data = arr_data.to_owned();
+            let arr_data = arr_data.into_shape(arr_len).unwrap();
+            let arr_data = arr_data.as_slice().unwrap();
+
+            let buffer = to_bytes(&arr_data, nb_bytes / dim0_size);
+            writer.write_all(&buffer).unwrap();
+        }
+    }
+}
+
+fn to_bytes<T>(data: &[T], size: usize) -> &[u8] {
```

Comment:
`safe-transmute` has a function for this: [`guarded_transmute_to_bytes_pod_many`](https://docs.rs/safe-transmute/0.8.0/safe_transmute/fn.guarded_transmute_to_bytes_pod_many.html)